content/200-concepts/100-components/02-prisma-client/260-client-extensions/130-result.mdx

```diff
@@ -9,8 +9,112 @@ tocDepth: 3
 
 <Admonition type="warning">
 
-We plan to add `result` extensions to a future version of Prisma. This page is a placeholder.
+This page is in draft.
 
 </Admonition>
 
+From version 4.7.0, you can use `result` [Prisma Client extensions](/concepts/components/prisma-client/client-extensions) to add custom fields and methods to query results.
+
+Use the `$extends` [client-level method](/reference/api-reference/prisma-client-reference#client-methods) to create an _extended client_. An extended client is a forked variant of the standard Prisma Client with an extension applied. Use the `results` extension type to add custom fields and methods to query results.
+
 </TopBlock>
+
+## Enable the preview feature
+
+Before you create Prisma Client extensions, you must enable the `clientExtensions` feature flag in the `generator` block of your `schema.prisma` file, as follows:
+
+```prisma
+generator client {
+  provider        = "prisma-client-js"
+  previewFeatures = ["clientExtensions"]
+}
+```
+
+## Add custom fields or methods to query results
+
+To add a custom [field](#add-a-custom-field-to-the-results-object) or [method](#add-a-custom-method-to-the-results-object) to query results, use the following structure. In this example, we extend query results in the `user` model.
+
+```ts
+const xprisma = prisma.$extends({
+  name?: 'name',
+  result?: {
+    user: {        // in this case, we extend query results in the user model
+      needs: { ... }
+      fields: { ... }
+    },
+  },
+});
+```
+
+- `name`: an optional field that you can use to name the extension. This name appears in error logs.
+- `result`: an optional field that defines new fields and methods to the query results.
+- `needs`: a required object which describes the dependencies of the result field.
+- `fields`: a required object that contains methods which defines how the virtual field is computed when it is accessed. You must declare equivelant objects in `fields` to the objects that you declare in `needs`.
+- `return`: the value of the computed field.
```

Comment:
```suggestion
    - `needs`: a required object which describes the dependencies of the result field.
    - `fields`: a required object that contains methods which defines how the virtual field is computed when it is accessed. Properties declared in `fields` must also be declared in `needs`.
```

Reply:
Actually, head up that the API can change, will post now the suggested changes

Reply:
Thanks - I've brought this change in